service/service.go

```diff
@@ -141,6 +146,39 @@ func New(ctx context.Context, set Settings, cfg Config) (*Service, error) {
 	return srv, nil
 }
 
+func Validate(ctx context.Context, set Settings, cfg Config) error {
```

Comment:
Why public?

Reply:
Because it's being called from the `otelcol` package.

Reply:
@ycombinator please take this as strong feedback that adding new public API to packages is discouraged at this time. Please add a godoc comment and prepare to discuss why this must be public.

Reply:
Got it, thanks for letting me know about trying not to increase the public API surface.  It makes sense.  Let me see if I can reimplement this functionality without adding to the public API, otherwise I will add a godoc here.

Reply:
Could we get the same result by just calling `New`? Is there a side effect to calling `New` which is being avoided by having a separate function?

Reply:
Thanks for the suggestion, @djaglowski - it totally works! Not sure why I didn't consider that much simpler approach 🤦.  Anyway, implemented in e373dd97f.